src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsModelValidatorProvider.cs

```diff
@@ -61,7 +61,11 @@ private static bool AddImplicitRequiredAttributeForValueTypes
                 {
                     factory = _defaultAttributeFactory;
                 }
-                results.Add(factory(attribute));
+                var modelValidator = factory(attribute);
+                if (!results.Exists(a => a.GetType() == modelValidator.GetType()))
+                {
+                    results.Add(factory(attribute));
+                }
```

Comment:
- why execute the delegate (`factory`) twice?
- is this a fix for a problem you found while testing?  should be done in a separate commit w/ clear comments in any case.